drivers/hmis/app/models/hmis/ce/referral_enroller.rb

```diff
@@ -48,28 +51,36 @@ def create_enrollment(message)
       referral.update!(target_enrollment: enrollment)
     end
 
-    # this is not actually an accessor method, even though RuboCop thinks it is
+    # Sets the Move-In Date on the target enrollments, based on a date value collected on the step form.
+    #
+    # This requires a Move-in date item to be on the step form with the following attributes:
+    #    { "link_id": "move_in_date", "type": "DATE", "mapping": { "custom_field_key": "<any field key for storing date value>" } }
+    #
+    # Note: We do a lookup by link_id rather than by mapping:field_name to prevent the FormProcessor from attempting
+    # to process the move_in_date field. That would be another approach, but we decided against it in favor of having an
+    # explicit trigger to populate the collected date onto the move_in_date field. In part because, when you go back and
+    # view a previously submitted task, it should show the Move-in Date as it was when the task was performed, NOT
+    # the current value of the Move-in Date field on the target enrollment. To achieve this, the date recorded on the task
+    # would be additionally stored in the CustomDataElement field.
     def set_move_in_date(message) # rubocop:disable Naming/AccessorMethodName
```

Comment:
This maybe should be a PR comment instead of a code comment, I can clean it up before merging.  I just want to call this out explicitly since it's a revert of a very recent change (https://github.com/open-path/Green-River/issues/7485). My bad on not really thinking ahead, here.

The previous approach would be to define the item as 
`{ "link_id": "whatever", "type": "DATE", "mapping": { "record_type": "ENROLLMENT", "field_name": "moveInDate" } }`.

Reply:
Thats clearer I will update, thank you for the suggestion!